src/MediawikiApiInterface.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Mediawiki\Api;
+
+/**
+ * @since 2.2
+ * @licence GNU GPL v2+
+ * @author Addshore
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+interface MediawikiApiInterface extends ApiRequester, AsyncApiRequester {
+
+	/**
+	 * @since 2.2
+	 *
+	 * @return bool|string false or the name of the current user
+	 */
+	public function isLoggedin();
+
+	/**
+	 * @since 2.2
+	 *
+	 * @param ApiUser $apiUser
+	 *
+	 * @throws UsageException
+	 * @return bool success
+	 */
+	public function login( ApiUser $apiUser );
```

Comment:
I think a AutoLoginMediawikiApi decorator or something similar might be a nice idea.